packages/network-controller/src/NetworkController.test.ts

```diff
@@ -1,278 +1,4132 @@
+                      },
+                      response: {
+                        result: '2',
+                      },
+                    },
+                  ]);
+                  createMetamaskProviderMock.mockReturnValue(
+                    fakeMetamaskProvider,
+                  );
+
+                  await waitForPublishedEvents(
+                    messenger,
+                    'NetworkController:providerConfigChange',
+                    {
+                      after: () => {
+                        controller.providerConfig = buildProviderConfig();
+                        assert(controller.provider);
+                      },
+                    },
+                  );
+
+                  await waitForStateChanges(messenger, {
+                    propertyPath: ['network'],
+                    count: 0,
+                    after: () => {
+                      controller.provider.emit('error', { some: 'error' });
+                    },
+                  });
+                  expect(controller.state.network).toBe('1');
+                },
+              );
+            });
+          });
+        });
+      });
+
+      describe('when the provider config in state contains a network type of "rpc"', () => {
+        describe('if the provider config contains an RPC target', () => {
+          it('sets the provider to a custom RPC provider initialized with the configured target, chain ID, nickname, and ticker', async () => {
+            await withController(
+              {
+                state: {
+                  providerConfig: {
+                    type: 'rpc',
+                    chainId: '123',
+                    nickname: 'some cool network',
+                    rpcTarget: 'http://example.com',
+                    ticker: 'ABC',
+                  },
+                },
+              },
+              ({ controller }) => {
+                const fakeMetamaskProvider = buildFakeMetamaskProvider();
+                createMetamaskProviderMock.mockReturnValue(
+                  fakeMetamaskProvider,
+                );
+
+                controller.providerConfig = buildProviderConfig({
+                  // NOTE: The type does not need to match the type in state
+                  type: 'mainnet',
+                });
+
+                expect(createMetamaskProviderMock).toHaveBeenCalledWith({
+                  type: 'mainnet',
+                  chainId: '123',
+                  engineParams: { pollingInterval: 12000 },
+                  nickname: 'some cool network',
+                  rpcUrl: 'http://example.com',
+                  ticker: 'ABC',
+                });
+                expect(controller.provider).toBe(fakeMetamaskProvider);
+              },
+            );
+          });
+
+          it('updates properties.isEIP1559Compatible in state based on the latest block (assuming that the request for eth_getBlockByNumber is made successfully)', async () => {
```

Comment:
Is this test a duplicate of the test on line 871 with the same name?

Reply:
It is, good catch. Removed in [a3f9afe](https://github.com/MetaMask/core/pull/1039/commits/a3f9afe0d06fae7a846e13d0239f724a0942fde8).